tests/ui/shadow.rs

```diff
@@ -1,6 +1,13 @@
+// aux-build:proc_macro_derive.rs
+
 #![warn(clippy::shadow_same, clippy::shadow_reuse, clippy::shadow_unrelated)]
 #![allow(clippy::let_unit_value)]
 
+extern crate proc_macro_derive;
+
+#[derive(proc_macro_derive::ShadowDerive)]
+pub struct Nothing;
+
```

Comment:
The `shadow*` lints already have a `ident.span.from_expansion()` guard and that is the reason why this snippet didn't trigger.

Which means that #9757 still needs investigation.